### ISSUE TITLE
Error in Console

### ISSUE THREAD

**Issue by Castiella** [BillyGalbreath/Ridables#34]
[12:32:09 ERROR]: Could not pass event CreatureSpawnEvent to Ridables v2.56

Any ideas how to fix this?

**Comment by Castiella** [comment]
at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80) ~[patched_1.13.2.jar:git-Paper-570]
at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:66) ~[patched_1.13.2.jar:git-Paper-570]
at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:520) ~[patched_1.13.2.jar:git-Paper-570]
at org.bukkit.craftbukkit.v1_13_R2.event.CraftEventFactory.callEvent(CraftEventFactory.java:104) ~[patched_1.13.2.jar:git-Paper-570]
at org.bukkit.craftbukkit.v1_13_R2.event.CraftEventFactory.callEntityDamageEvent(CraftEventFactory.java:834) ~[patched_1.13.2.jar:git-Paper-570]
at org.bukkit.craftbukkit.v1_13_R2.event.CraftEventFactory.handleEntityDamageEvent(CraftEventFactory.java:733) ~[patched_1.13.2.jar:git-Paper-570]
at org.bukkit.craftbukkit.v1_13_R2.event.CraftEventFactory.handleLivingEntityDamageEvent(CraftEventFactory.java:866) ~[patched_1.13.2.jar:git-Paper-570]
at net.minecraft.server.v1_13_R2.EntityLiving.damageEntity0(EntityLiving.java:1520) ~[patched_1.13.2.jar:git-Paper-570]
at net.minecraft.server.v1_13_R2.EntityLiving.damageEntity(EntityLiving.java:993) ~[patched_1.13.2.jar:git-Paper-570]
at net.minecraft.server.v1_13_R2.EntityMonster.damageEntity(EntityMonster.java:44) ~[patched_1.13.2.jar:git-Paper-570]
at net.minecraft.server.v1_13_R2.EntityEnderman.damageEntity(EntityEnderman.java:267) ~[patched_1.13.2.jar:git-Paper-570]
at net.minecraft.server.v1_13_R2.EntityHuman.attack(EntityHuman.java:1111) ~[patched_1.13.2.jar:git-Paper-570]
at net.minecraft.server.v1_13_R2.EntityPlayer.attack(EntityPlayer.java:1446) ~[patched_1.13.2.jar:git-Paper-570]
at net.minecraft.server.v1_13_R2.PlayerConnection.a(PlayerConnection.java:2010) ~[patched_1.13.2.jar:git-Paper-570]
at net.minecraft.server.v1_13_R2.PacketPlayInUseEntity.a(PacketPlayInUseEntity.java:49) ~[patched_1.13.2.jar:git-Paper-570]
at net.minecraft.server.v1_13_R2.PacketPlayInUseEntity.a(PacketPlayInUseEntity.java:6) ~[patched_1.13.2.jar:git-Paper-570]

**Comment by BillyGalbreath** [comment]
Can I have more of the error/log for this one? What's there doesn't actually contain any error

**Comment by Castiella** [comment]
Its literally spamming the console, here you go :) 
[latest (7).log](https://github.com/BillyGalbreath/Ridables/files/2993011/latest.7.log)

**Comment by Jikoo** [comment]
Spigot [updated mappings again](https://hub.spigotmc.org/stash/projects/SPIGOT/repos/craftbukkit/commits/a49270b28b35c0b110cfa5fe0f03c9621a899d80).

```
[03:04:42] [Server thread/INFO]: [Ridables] Loading Ridables v2.56
[03:04:42] [Server thread/INFO]: [Ridables] Skipping disabled entity: bat
[03:04:42] [Server thread/INFO]: [Ridables] Skipping disabled entity: blaze
[03:04:42] [Server thread/INFO]: [Ridables] Skipping disabled entity: cave_spider
[03:04:42] [Server thread/INFO]: [Ridables] Skipping disabled entity: chicken
[03:04:42] [Server thread/INFO]: [Ridables] Skipping disabled entity: cod
[03:04:42] [Server thread/INFO]: [Ridables] Skipping disabled entity: cow
[03:04:42] [Server thread/INFO]: [Ridables] Skipping disabled entity: creeper
[03:04:42] [Server thread/INFO]: [Ridables] Successfully injected replacement entity: dolphin
[03:04:42] [Server thread/INFO]: [Ridables] Skipping disabled entity: drowned
[03:04:42] [Server thread/INFO]: [Ridables] Skipping disabled entity: elder_guardian
[03:04:42] [Server thread/INFO]: [Ridables] Skipping disabled entity: ender_dragon
[03:04:42] [Server thread/INFO]: [Ridables] Skipping disabled entity: enderman
[03:04:42] [Server thread/INFO]: [Ridables] Skipping disabled entity: endermite
[03:04:42] [Server thread/INFO]: [Ridables] Skipping disabled entity: evoker
[03:04:42] [Server thread/INFO]: [Ridables] Skipping disabled entity: ghast
[03:04:42] [Server thread/INFO]: [Ridables] Skipping disabled entity: giant
[03:04:42] [Server thread/INFO]: [Ridables] Skipping disabled entity: guardian
[03:04:42] [Server thread/INFO]: [Ridables] Skipping disabled entity: husk
[03:04:42] [Server thread/INFO]: [Ridables] Skipping disabled entity: illusioner
[03:04:42] [Server thread/INFO]: [Ridables] Skipping disabled entity: iron_golem
[03:04:42] [Server thread/INFO]: [Ridables] Skipping disabled entity: llama
[03:04:42] [Server thread/INFO]: [Ridables] Skipping disabled entity: magma_cube
[03:04:42] [Server thread/INFO]: [Ridables] Skipping disabled entity: mooshroom
[03:04:42] [Server thread/INFO]: [Ridables] Skipping disabled entity: ocelot
[03:04:42] [Server thread/INFO]: [Ridables] Skipping disabled entity: parrot
[03:04:42] [Server thread/INFO]: [Ridables] Skipping disabled entity: phantom
[03:04:42] [Server thread/ERROR]: java.lang.IllegalAccessError: no such constructor: net.pl3x.bukkit.ridables.entity.RidablePig.<init>(World)void/newInvokeSpecial initializing Ridables v2.56 (Is it up to date?)
java.lang.BootstrapMethodError: java.lang.IllegalAccessError: no such constructor: net.pl3x.bukkit.ridables.entity.RidablePig.<init>(World)void/newInvokeSpecial
        at net.pl3x.bukkit.ridables.entity.RidableType.<clinit>(RidableType.java:54) ~[?:?]
        at net.pl3x.bukkit.ridables.Ridables.onLoad(Ridables.java:73) ~[?:?]
        at org.bukkit.craftbukkit.v1_13_R2.CraftServer.loadPlugins(CraftServer.java:330) [spigot.jar:git-Spigot-e6eb36f-d3ed151]
        at net.minecraft.server.v1_13_R2.DedicatedServer.init(DedicatedServer.java:213) [spigot.jar:git-Spigot-e6eb36f-d3ed151]
        at net.minecraft.server.v1_13_R2.MinecraftServer.run(MinecraftServer.java:698) [spigot.jar:git-Spigot-e6eb36f-d3ed151]
        at java.lang.Thread.run(Thread.java:745) [?:1.8.0_101]
Caused by: java.lang.IllegalAccessError: no such constructor: net.pl3x.bukkit.ridables.entity.RidablePig.<init>(World)void/newInvokeSpecial
        at java.lang.invoke.MethodHandleNatives.linkMethodHandleConstant(MethodHandleNatives.java:483) ~[?:1.8.0_101]
        ... 6 more
Caused by: java.lang.NoClassDefFoundError: net/minecraft/server/v1_13_R2/BlockPosition$b
        at java.lang.invoke.MethodHandleNatives.resolve(Native Method) ~[?:1.8.0_101]
        at java.lang.invoke.MemberName$Factory.resolve(MemberName.java:975) ~[?:1.8.0_101]
        at java.lang.invoke.MemberName$Factory.resolveOrFail(MemberName.java:1000) ~[?:1.8.0_101]
        at java.lang.invoke.MethodHandles$Lookup.resolveOrFail(MethodHandles.java:1389) ~[?:1.8.0_101]
        at java.lang.invoke.MethodHandles$Lookup.linkMethodHandleConstant(MethodHandles.java:1745) ~[?:1.8.0_101]
        at java.lang.invoke.MethodHandleNatives.linkMethodHandleConstant(MethodHandleNatives.java:477) ~[?:1.8.0_101]
        ... 6 more
Caused by: java.lang.ClassNotFoundException: net.minecraft.server.v1_13_R2.BlockPosition$b
        at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:87) ~[spigot.jar:git-Spigot-e6eb36f-d3ed151]
        at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:82) ~[spigot.jar:git-Spigot-e6eb36f-d3ed151]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:424) ~[?:1.8.0_101]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357) ~[?:1.8.0_101]
        at java.lang.invoke.MethodHandleNatives.resolve(Native Method) ~[?:1.8.0_101]
        at java.lang.invoke.MemberName$Factory.resolve(MemberName.java:975) ~[?:1.8.0_101]
        at java.lang.invoke.MemberName$Factory.resolveOrFail(MemberName.java:1000) ~[?:1.8.0_101]
        at java.lang.invoke.MethodHandles$Lookup.resolveOrFail(MethodHandles.java:1389) ~[?:1.8.0_101]
        at java.lang.invoke.MethodHandles$Lookup.linkMethodHandleConstant(MethodHandles.java:1745) ~[?:1.8.0_101]
        at java.lang.invoke.MethodHandleNatives.linkMethodHandleConstant(MethodHandleNatives.java:477) ~[?:1.8.0_101]
        ... 6 more
```
All errors point to the same NoClassDefFound:
```
Caused by: java.lang.NoClassDefFoundError: Could not initialize class net.pl3x.bukkit.ridables.entity.RidableType
        at net.pl3x.bukkit.ridables.listener.RidableListener.onCreatureSpawn(RidableListener.java:60) ~[?:?]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_101]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_101]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_101]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_101]
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:312) ~[spigot.jar:git-Spigot-e6eb36f-d3ed151]
        ... 29 more
```

**Comment by BillyGalbreath** [comment]
Fixed in [v3.0](https://github.com/BillyGalbreath/Ridables/releases/tag/v3.0)
